### DISCUSSION TITLE
Smart Plug Config_overtemp_enable and dnd switchs

### DISCUSSION THREAD

**Discussion by vinceg0267** [krahabb/meross_lan#347] [Q&A]
Hi, What are these switches for?

**Comment by krahabb** [comment]
Hello @vinceg0267,
the DND entity repsents the 'do not disturb' state of the device and is usually associated with some 'presence light' on the device.  By turning it off the device would stop emitting any presence light (like representing the switch state) so to not pollute the environment with 'disturbing' lightning 

the 'overtemp enable' is a feature appearing on latest device families (I've recovered that from a diagnostic trace submitted by a meross_lan user) and I don't have a clue about its meaning. It should be something like a safety feature you can enable on the device in order to shut it down when it is overheating but I'm waiting to know about how it works and if it might be useful from people actually using it on the Meross app.
Together with the 'overtemp enable' switch there should be a binary sensor reporting the actual state of the overtemp

**Reply by vinceg0267** [reply]
Thanks for the explanation @krahabb. Yes, indeed there is also a binary sensor called config_overtemp_type

**Reply by bradkemper** [reply]
I think I clicked my Config_overtemp_enable switch a couple times by accident. Is it normally supposed to be on, or off? I.e., do you think it turns on/off a temperature sensor to prevent overheating? Or does it enable overheating, to bypass an automatic shut off (like for diagnosing problems)? I mean, it sounds like the latter, enabling overheating, but it could mean enabling an overheating safety measure.

**Reply by krahabb** [reply]
I don't really know to be honest, since it just appeared in some traces from new devices here and there so I've preemptively added the switch in order to be already prepared to manage it in HA should someone managed to understand its behavior.

The switch is not the smartest solution though since those traces carry by default the value '-1' which is then mapped to the 'on' state in HA terms.
But, if you toggle it in HA, it will be set to '1' (for 'on' and '0' for 'off') which might be different in meaning than '-1'.

I think the best option would be to identify the feature in the Meross app and manage it there (seeing then if the HA UI replies the new setting).

**Reply by markcst** [reply]
> Config_overtemp_enable

So, no clue about this? I'm wondering too what is it

**Comment by rknightion** [comment]
FYI @krahabb & @markcst the config overtemp enable item is if the overheat sensor is enabled on the MS power plugs. It has different 'types' with mode 1 meaning enabled with alert functionality but mode 2 will actually power down the device if a overtemp event happens.

I don't have them on my older MS2xx switches but do on my 315s.
@bradkemper by default it should be enabled and set to alert only.

<img width="200" alt="IMG_5242" src="https://github.com/user-attachments/assets/e707fddf-8cfa-404f-b1e6-2981975353e2" />

**Reply by krahabb** [reply]
Thank you @rknightion,
I'm going to implement a better entity management based off this.
Btw, do you know what the protocol value is for disabled ? It should 'naturally' be '0' but as far as I've seen in traces, sometimes a '-1' is carried.

**Reply by saranity** [reply]
Would it be possible to also make the over temperature alert visible in HA?
